### PR TITLE
Respect cull face during shadow pass

### PR DESCRIPTION
**Pull request by HifiExperiments**
https://github.com/overte-org/overte/issues/547

### CODE REVIEW

File: libraries/render-utils/src/RenderPipelines.cpp [hunk 1]
```diff
@@ -46,168 +48,164 @@ void initZPassPipelines(ShapePlumber& plumber, gpu::StatePointer state, const re
...
 void lightBatchSetter(const ShapePipeline& pipeline, gpu::Batch& batch, RenderArgs* args);
 static bool forceLightBatchSetter{ false };
 
-void initDeferredPipelines(render::ShapePlumber& plumber, const render::ShapePipeline::BatchSetter& batchSetter, const render::ShapePipeline::ItemSetter& itemSetter) {
-    using namespace shader::render_utils::program;
-    using Key = render::ShapeKey;
-    auto addPipeline = std::bind(&addPlumberPipeline, std::ref(plumber), _1, _2, _3, _4);
-
-    // TOOD: build this list algorithmically so we don't have to maintain it
-    std::vector<std::pair<render::ShapeKey::Builder, uint32_t>> pipelines = {
```
Comment (HifiExperiments) [inline libraries/render-utils/src/RenderPipelines.cpp:62]: due to some complexities of how our ShapeKeys work, I found it simpler to combine the deferred and shadow pipeline maps into one global shared list, even though it means repeating the shadow pipelines.  this allows us to use `addPipeline`/`addPlumberPipeline` (as opposed to `shapePlumber.addPipeline`) in `initZPassPipelines` below.

**Review by daleglass — APPROVED** [review]
So I looked this over, but all I can say about it is that the C++ looks okay to me.

I can't really say much about renderers, so hopefully somebody else can chime in on that.

**Comment by ksuprynowicz** [comment]
I tested it with model having both single and double sided materials and everything works perfectly.

**Comment by ksuprynowicz** [comment]
Can we merge this?

**Comment by HifiExperiments** [comment]
yes!

**Pull request merged**